packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -599,3 +601,105 @@ export enum UsageFlags {
   // Input will also be set when MultipartFormData is set.
   MultipartFormData = 1 << 5,
 }
+
+export type SdkExampleBase = {
+  kind: string;
+  name: string;
+  description: string;
+  filePath: string;
+  rawExample: any;
+};
+
+export interface SdkHttpOperationExample extends SdkExampleBase {
+  kind: "http";
+  parameters: SdkHttpParameterExample[];
+  responses: Map<number, SdkHttpResponseExample>;
+}
+
+export interface SdkHttpParameterExample {
+  parameter: SdkHttpParameter;
+  value: SdkTypeExample;
+}
+
+export interface SdkHttpResponseExample {
+  response: SdkHttpResponse;
+  headers: SdkHttpResponseHeaderExample[];
+  value?: SdkTypeExample;
+}
+
+export interface SdkHttpResponseHeaderExample {
+  header: SdkServiceResponseHeader;
+  value: SdkTypeExample;
+}
+
+export type SdkTypeExample =
+  | SdkStringExample
+  | SdkNumberExample
+  | SdkBooleanExample
+  | SdkNullExample
+  | SdkAnyExample
+  | SdkArrayExample
+  | SdkDictionaryExample
+  | SdkUnionExample
+  | SdkModelExample;
+
+export interface SdkExampleTypeBase {
+  kind: string;
+  type: SdkType | SdkModelPropertyType;
+  value: unknown;
+}
+
+export interface SdkStringExample extends SdkExampleTypeBase {
+  kind: "string";
+  type: SdkBuiltInType | SdkDateTimeType | SdkDurationType | SdkEnumType | SdkConstantType;
+  value: string;
```

Comment:
the type here does not really make sense, because builtintype also includes the numbers.

Could this work?
```suggestion
  type: SdkBuiltInType & { kind: "string" } | SdkDateTimeType | SdkDurationType | SdkEnumType | SdkConstantType;
  value: string;
```

Reply:
same for the numbers and boolean.

Reply:
if we have my PR on scalars merged, here could be more clear because those based on string types will always have the kind of `string` (such as `azurelocation`)

Reply:
i just want to list all possible types here. i believe i could have some optimization after your pr merged.